### PR TITLE
Enum now accepts String literals and final ...

### PR DESCRIPTION
**Pull request by vincent-prz**
This aims to solve https://github.com/python/mypy/issues/8219

This PR is still a draft, but am I going in the right direction ? I am doing lot of if refinements from line 150, in order to be compliant with mypy linting, maybe there is a less clunky way to achieve the same result ? Also, should the following cases typecheck ?

```python
class A:
    x: Literal['x y'] = 'x y'

Coordinates = Enum('Coordinates', A.x)
```

```python
def f() -> Literal['x y']
    return 'x y'

Coordinates = Enum('Coordinates', f())
```

Moreover, I had to add `[builtins fixtures/tuple.pyi]` in my tests, they would otherwise crash, but I am not sure to understand why.

### CODE REVIEW

File: mypy/semanal_enum.py [hunk 1]
```diff
@@ -146,6 +147,25 @@ def parse_enum_call_args(self, call: CallExpr,
...
+                fields = proper_type.value
+                for field in fields.replace(',', ' ').split():
+                    items.append(field)
+            elif args[1].node.is_final and isinstance(args[1].node.final_value, str):
+                fields = args[1].node.final_value
+                for field in fields.replace(',', ' ').split():
+                    items.append(field)
+            else:
+                return self.fail_enum_call_arg(
+                    "%s() expects a string, tuple, list or dict literal as the second argument" %
```
Comment (97littleleaf11) [inline mypy/semanal_enum.py:166]: Could you please merge the same error messages into messages_registry.py

**Review by 97littleleaf11 — COMMENTED** [review]
@sobolevn Do you have any suggestions on this PR?

File: test-data/unit/check-newsemanal.test [hunk 2]
```diff
@@ -2058,6 +2058,24 @@ from enum import Enum
...
+Animal = Enum('Animal', x)
+
+[builtins fixtures/tuple.pyi]
+
+[case testNewAnalyzerEnumAcceptsFinalValue]
+from enum import Enum
+from typing_extensions import Final
+
+x: Final['str'] = 'ANT BEE CAT DOG'
+Animal = Enum('Animal', x)
```
Comment (sobolevn) [inline test-data/unit/check-newsemanal.test:2075]: I would also check that `Animal.ANT`, `Animal.BEE`, etc are available in both test cases.

**Review by sobolevn — COMMENTED** [review]
Sorry, I've missed it! Thanks for the ping 🙂

**Comment by 97littleleaf11** [comment]
cc @sobolevn

**Pull request merged**